ui/app/registration/controllers/editPatientController.js

```diff
@@ -97,8 +97,49 @@ angular.module('bahmni.registration')
                 return $scope.readOnlyFields ? ($scope.readOnlyFields[field] ? true : false) : undefined;
             };
 
+            $scope.sendWhatsAppMessage = function () {
+                var whatsAppMessage = appService.getAppDescriptor().getConfigValue("whatsAppMessage") || $translate.instant(Bahmni.Registration.Constants.whatsAppMessage);
+                var startIndex = whatsAppMessage.indexOf("{");
+                var endIndex = whatsAppMessage.indexOf("}");
+                var messageSubstring = whatsAppMessage.substring(startIndex, endIndex + 1);
+                var patientAttributes = messageSubstring.substring(1, messageSubstring.length - 1).split(",");
```

Comment:
sorry I should have caught this earlier... This code is too verbose and its hard to understand what you trying to do. Basically you just need to interpolate values into your translation text - like a string template. We could simply use replace function in javascript without the need to have the imperative loops and WET code (see 110-112, 116-118, 122-123 etc are just same)...

```
var name = 'n0man'
var age = 99
var message = "you are not allowed to work #name because you are #age"

console.log(message.replace('#name', name).replace('#age', age))
```

Reply:
I agree. In different languages the location of the words / brackets changes.. so using template messages like @N0-man mentioned is more robust.

Reply:
Updated the message configuration  with replace function instead of loops.